docs/en/integrations/data-visualization/grafana/index.md

```diff
@@ -0,0 +1,103 @@
+
+To configure a read-only user, follow these steps:
+1. Create a `readonly` user profile following the [Creating Users and Roles in ClickHouse](/docs/en/operations/access-rights) guide.
+2. Ensure the `readonly` user has enough permission to modify the `max_execution_time` setting required by the underlying [clickhouse-go client](https://github.com/ClickHouse/clickhouse-go).
+3. If you're using a public ClickHouse instance, it is not recommended to set `readonly=2` in the `readonly` profile. Instead, leave `readonly=1` and set the constraint type of `max_execution_time` to [changeable_in_readonly](/docs/en/operations/settings/constraints-on-settings) to allow modification of this setting.
+
+## 3.  Install the ClickHouse Plugin for Grafana
+
+Before Grafana can connect to ClickHouse, you need to install the appropriate Grafana plugin. Assuming you are logged in to Grafana, follow these steps:
+
+1. From the **Connections** page in the sidebar, select the **Add new connection** tab.
+
+2. Search for **ClickHouse** and click on the signed plugin by Grafana Labs:
+
+    <img src={require('./images/search.png').default} class="image" alt="Select the ClickHouse plugin on the connections page" />
+
+3. On the next screen, click the **Install** button:
+
+    <img src={require('./images/install.png').default} class="image" alt="Install the ClickHouse plugin" />
+
+## 4. Define a ClickHouse data source
+
+1. Once the installation is complete, click the **Add new data source** button. (You can also add a data source from the **Data sources** tab on the **Connections** page.)
+
+    <img src={require('./images/add_new_ds.png').default} class="image" alt="Create a ClickHouse data source" />
+
+2. Either scroll down and find the **ClickHouse** data source type, or you can search for it in the search bar of the **Add data source** page. Select the **ClickHouse** data source and the following page will appear:
+
+  <img src={require('./images/quick_config.png').default} class="image" alt="Connection configuration page" />
+
+1. Enter your server settings and credentials. The key settings are:
+
+- **Server host address:** the hostname of your ClickHouse service.
+- **Server port:** the port for your ClickHouse service. Will be different depending on server configuration and protocol.
+- **Protocol** the protocol used to connect to your ClickHouse service.
+- **Secure connection** enable if your server requires a secure connection.
+- **Username** and **Password**: enter your ClickHouse user credentials. If you have not configured any users, try `default` for the username. It is recommended to [configure a read-only user](#2-making-a-read-only-user).
+
+For more settings, check the [plugin configuration](./config.md) documentation.
+
+1. Click the **Save & test** button to verify that Grafana can connect to your ClickHouse service. If successful, you will see a **Data source is working** message:
```

Comment:
Enumeration is broken
![2024-02-21_16-43-40](https://github.com/ClickHouse/clickhouse-docs/assets/3198181/a41c277d-51fa-4e43-9cd7-afed96dfdbb9)